net/core/filter.c

```diff
@@ -5849,6 +5855,278 @@ static const struct bpf_func_proto bpf_tcp_gen_syncookie_proto = {
+};
+
+BPF_CALL_2(bpf_lua_dataunref, struct xdp_buff *, ctx, int, data_ref) {
+	verify_and_lock();
+	ldata_unref(ctx->L, data_ref);
+	return 0;
+}
+
+static const struct bpf_func_proto bpf_lua_dataunref_proto = {
+	.func		= bpf_lua_dataunref,
+	.gpl_only	= false,
+	.pkt_access	= false,
+	.ret_type	= RET_VOID,
+	.arg1_type	= ARG_PTR_TO_CTX,
+	.arg2_type	= ARG_ANYTHING,
+};
+
+BPF_CALL_4(bpf_lua_pcall, struct xdp_buff *, ctx, char *, funcname,
+			int, num_args, int, num_rets) {
+	int base;
+
+	verify_and_lock();
+
+	base = lua_gettop(ctx->L) - num_args;
+	if (lua_getglobal(ctx->L, funcname) != LUA_TFUNCTION) {
+		pr_err("function %s not found\n", funcname);
+		num_rets = 0;
+		goto clean_state;
+	}
+
+	lua_insert(ctx->L, base + 1);
+	if (lua_pcall(ctx->L, num_args, num_rets, 0)) {
+		pr_err("%s\n", lua_tostring(ctx->L, -1));
+		num_rets = 0;
+		goto clean_state;
+	}
+
+	base += num_rets;
+
+clean_state:
+	lua_settop(ctx->L, base);
+	return num_rets;
+}
+
+static const struct bpf_func_proto bpf_lua_pcall_proto = {
+	.func		= bpf_lua_pcall,
+	.gpl_only	= false,
+	.pkt_access	= false,
+	.ret_type	= RET_INTEGER,
+	.arg1_type	= ARG_PTR_TO_CTX,
+	.arg2_type	= ARG_ANYTHING,
+	.arg3_type	= RET_INTEGER,
+	.arg4_type	= RET_INTEGER,
+};
+
+BPF_CALL_2(bpf_lua_pop, struct xdp_buff *, ctx, int, index) {
+	verify_and_lock();
+	lua_pop(ctx->L, index);
+	return 0;
+}
+
+static const struct bpf_func_proto bpf_lua_pop_proto = {
+	.func		= bpf_lua_pop,
+	.gpl_only	= false,
+	.pkt_access	= false,
+	.ret_type	= RET_VOID,
+	.arg1_type	= ARG_PTR_TO_CTX,
+	.arg2_type	= ARG_ANYTHING,
+};
+
+BPF_CALL_2(bpf_lua_pushinteger, struct xdp_buff *, ctx, int, num) {
+	verify_and_lock();
+	lua_pushinteger(ctx->L, num);
+	return 0;
+}
+
+static const struct bpf_func_proto bpf_lua_pushinteger_proto = {
+	.func		= bpf_lua_pushinteger,
+	.gpl_only	= false,
+	.pkt_access	= false,
+	.ret_type	= RET_VOID,
+	.arg1_type	= ARG_PTR_TO_CTX,
+	.arg2_type	= ARG_ANYTHING,
+};
+
+BPF_CALL_2(bpf_lua_pushlightuserdata, struct xdp_buff *, ctx, void *, ptr) {
+	verify_and_lock();
+	lua_pushlightuserdata(ctx->L, ptr);
+	return 0;
+}
+
+static const struct bpf_func_proto bpf_lua_pushlightuserdata_proto = {
+	.func		= bpf_lua_pushlightuserdata,
+	.gpl_only	= false,
+	.pkt_access	= false,
+	.ret_type	= RET_VOID,
+	.arg1_type	= ARG_PTR_TO_CTX,
+	.arg2_type	= ARG_ANYTHING,
+};
+
+BPF_CALL_2(bpf_lua_pushmap, struct xdp_buff *, ctx, struct bpf_map *, map) {
+	verify_and_lock();
+	lua_pushlightuserdata(ctx->L, map);
+	return 0;
+}
+
+static const struct bpf_func_proto bpf_lua_pushmap_proto = {
+	.func		= bpf_lua_pushmap,
+	.gpl_only	= false,
+	.pkt_access	= false,
+	.ret_type	= RET_VOID,
+	.arg1_type	= ARG_PTR_TO_CTX,
+	.arg2_type	= ARG_CONST_MAP_PTR,
+};
+
+BPF_CALL_3(bpf_lua_pushlstring, struct xdp_buff *, ctx, const char *, str, size_t, len) {
+	verify_and_lock();
+	lua_pushlstring(ctx->L, str, len);
+	return 0;
+}
+
+static const struct bpf_func_proto bpf_lua_pushlstring_proto = {
```

Comment:
I really believe we can have a macro to generate the wrapper.. including such struct..